src/main/java/com/netflix/conductor/kafka/eventqueue/KafkaConsumeObservableQueue.java

```diff
@@ -0,0 +1,316 @@
+            }
+            prop = properties.getOffset();
+            if (prop != null) {
+                consumerProperties.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, prop);
+            }
+            //consumerProperties.put("security.protocol", SecurityProtocol.SASL_PLAINTEXT.name);
+            consumerProperties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+            consumerProperties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+            consumerProperties.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, false);
+
+            consumerProperties.put(ConsumerConfig.CLIENT_ID_CONFIG, queueURI + "_conductor_consumer");
+            checkConsumerProps(consumerProperties);
+
+            this.consumer = new KafkaConsumer<>(consumerProperties);
+            this.consumer.subscribe(this.topics);
+        } catch (KafkaException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Override
+    public String getName() {
+        return queueURI;
+    }
+
+    @Override
+    public String getURI() {
+        return queueURI;
+    }
+
+    /**
+     * Checks mandatory configs are available for kafka consumer.
+     */
+    private void checkConsumerProps(Properties consumerProps) {
+        List<String> mandatoryKeys = Arrays.asList(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG,
+                ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG);
+        List<String> keysNotFound = hasKeyAndValue(consumerProps, mandatoryKeys);
+        if (keysNotFound.size() > 0) {
+            String error = String.format("Configuration missing for Kafka consumer. %s", keysNotFound);
+            logger.error(error);
+            throw new RuntimeException(error);
+        }
+    }
+
+    /**
+     * Validates whether the property has given keys.
+     */
+    private List<String> hasKeyAndValue(Properties prop, List<String> keys) {
+        List<String> keysNotFound = new ArrayList<>();
+        for (String key : keys) {
+            if (!prop.containsKey(key) || Objects.isNull(prop.get(key))) {
+                keysNotFound.add(key);
+            }
+        }
+        return keysNotFound;
+    }
+
+    @Override
+    public Observable<Message> observe() {
+        Observable.OnSubscribe<Message> subscriber = getOnSubscribe();
+        return Observable.create(subscriber);
+    }
+
+    private Observable.OnSubscribe<Message> getOnSubscribe() {
+        return subscriber -> {
+            Observable<Long> interval = Observable.interval(pollIntervalInMS, TimeUnit.MILLISECONDS);
+            interval.flatMap((Long x) -> Observable.from(receiveMessages()))
+                    .subscribe(subscriber::onNext, subscriber::onError);
+        };
+    }
+
+    @Override
+    public List<String> ack(List<Message> messages) {
+        List<String> messageIds = new ArrayList<>();
+        for (Message message : messages) {
```

Comment:
This logic still have a problem. What if message with offset "2" is not in the "mesages" list but message with offset "3" is here?